File: ROS2/Day1/Day1_ROS2_package/changgyu_task2_py/changgyu_task2_py/publisher.py
import rclpy
from rclpy.node import Node

from std_msgs.msg import Int32
from std_msgs.msg import String
from std_msgs.msg import Float32
from std_msgs.msg import Bool


class Task2Publisher(Node):

    def __init__(self):
        super().__init__('task2_py_publisher')

        self.int_publisher = self.create_publisher(
            Int32, '/task2/int_data', 10
        )

        self.string_publisher = self.create_publisher(
            String, '/task2/string_data', 10
        )

        self.float_publisher = self.create_publisher(
            Float32, '/task2/float_data', 10
        )

        self.bool_publisher = self.create_publisher(
            Bool, '/task2/bool_data', 10
        )

        self.timer = self.create_timer(
            1.0, self.publish_task2_msg
        )

        self.count = 0

    def publish_task2_msg(self):
        int_msg = Int32()
        string_msg = String()
        float_msg = Float32()
        bool_msg = Bool()

        int_msg.data = self.count
        string_msg.data = 'Hello ROS2 Python'
        float_msg.data = 3.14
        bool_msg.data = (self.count % 2 == 0)

        self.int_publisher.publish(int_msg)
        self.string_publisher.publish(string_msg)
        self.float_publisher.publish(float_msg)
        self.bool_publisher.publish(bool_msg)

        self.get_logger().info(
            'Publish -> int: {0}, string: {1}, float: {2:.2f}, bool: {3}'.format(
                int_msg.data,
                string_msg.data,
                float_msg.data,
                bool_msg.data
            )
        )

        self.count += 1


def main(args=None):
    rclpy.init(args=args)

    node = Task2Publisher()

    try:
        rclpy.spin(node)

    except KeyboardInterrupt:
        node.get_logger().info('Keyboard Interrupt (SIGINT)')

    finally:
        node.destroy_node()
        rclpy.shutdown()


if __name__ == '__main__':
    main()

File: ROS2/Day1/Day1_ROS2_package/changgyu_task2_cpp/src/publisher.cpp
#include "changgyu_task2_cpp/publisher.hpp"

#include <chrono>
#include <functional>

using namespace std::chrono_literals;

Task2Publisher::Task2Publisher()
    : Node("task2_cpp_publisher"), count_(0)
{
    int_publisher_ =
        this->create_publisher<std_msgs::msg::Int32>(
            "/task2/int_data", 10);

    string_publisher_ =
        this->create_publisher<std_msgs::msg::String>(
            "/task2/string_data", 10);

    float_publisher_ =
        this->create_publisher<std_msgs::msg::Float32>(
            "/task2/float_data", 10);

    bool_publisher_ =
        this->create_publisher<std_msgs::msg::Bool>(
            "/task2/bool_data", 10);

    timer_ = this->create_wall_timer(
        1s,
        std::bind(&Task2Publisher::timer_callback, this));
}

void Task2Publisher::timer_callback()
{
    auto int_msg = std_msgs::msg::Int32();
    auto string_msg = std_msgs::msg::String();
    auto float_msg = std_msgs::msg::Float32();
    auto bool_msg = std_msgs::msg::Bool();

    int_msg.data = count_;
    string_msg.data = "Hello ROS2 C++";
    float_msg.data = 3.14f;
    bool_msg.data = (count_ % 2 == 0);

    int_publisher_->publish(int_msg);
    string_publisher_->publish(string_msg);
    float_publisher_->publish(float_msg);
    bool_publisher_->publish(bool_msg);

    RCLCPP_INFO(
        this->get_logger(),
        "Publish -> int: %d, string: %s, float: %.2f, bool: %s",
        int_msg.data,
        string_msg.data.c_str(),
        float_msg.data,
        bool_msg.data ? "true" : "false");

    count_++;
}

int main(int argc, char *argv[])
{
    rclcpp::init(argc, argv);

    auto node = std::make_shared<Task2Publisher>();
    rclcpp::spin(node);

    rclcpp::shutdown();

    return 0;
}